diff --git a/node_modules/gatsby/dist/internal-plugins/functions/gatsby-node.js b/node_modules/gatsby/dist/internal-plugins/functions/gatsby-node.js
index 2839482..a2e2dfa 100644
--- a/node_modules/gatsby/dist/internal-plugins/functions/gatsby-node.js
+++ b/node_modules/gatsby/dist/internal-plugins/functions/gatsby-node.js
@@ -16,6 +16,8 @@ var _glob = _interopRequireDefault(require("glob"));
 
 var _path2 = _interopRequireDefault(require("path"));
 
+var _crypto = _interopRequireDefault(require("crypto"));
+
 var _webpack = _interopRequireDefault(require("webpack"));
 
 var _gatsbyCoreUtils = require("gatsby-core-utils");
@@ -225,6 +227,17 @@ const createWebpackConfig = async ({
   }, {
     "process.env": `({})`
   });
+
+  const envVarHash = _crypto.default.createHash(`md5`).update(JSON.stringify(processEnvVars)).digest(`hex`);
+  /**
+   * Invalidate webpack cache on env var value change.
+   * @see {@link https://webpack.js.org/plugins/define-plugin/#runtime-values-via-runtimevalue}
+   */
+
+
+  processEnvVars[`ENV_VAR_HASH`] = _webpack.default.DefinePlugin.runtimeValue(() => envVarHash, {
+    version: () => envVarHash
+  });
   const entries = {};
   const precompileDevFunctions = isProductionEnv || process.env.GATSBY_PRECOMPILE_DEVELOP_FUNCTIONS === `true` || process.env.GATSBY_PRECOMPILE_DEVELOP_FUNCTIONS === `1`;
   const functionsList = precompileDevFunctions ? knownFunctions : activeDevelopmentFunctions;
